x-pack/plugins/lens/public/xy_visualization/xy_config_panel/color_picker.tsx

```diff
@@ -47,49 +42,64 @@ export const ColorPicker = ({
   disableHelpTooltip,
   disabled,
   setConfig,
+  showAlpha,
+  defaultColor,
 }: VisualizationDimensionEditorProps<State> & {
   formatFactory: FormatFactory;
   paletteService: PaletteRegistry;
   label?: string;
   disableHelpTooltip?: boolean;
   disabled?: boolean;
   setConfig: (config: { color?: string }) => void;
+  showAlpha?: boolean;
+  defaultColor?: string;
 }) => {
   const index = state.layers.findIndex((l) => l.layerId === layerId);
   const layer = state.layers[index];
 
   const overwriteColor = getSeriesColor(layer, accessor);
   const currentColor = useMemo(() => {
     if (overwriteColor || !frame.activeData) return overwriteColor;
-    if (isReferenceLayer(layer)) {
-      return defaultReferenceLineColor;
-    } else if (isAnnotationsLayer(layer)) {
-      return defaultAnnotationColor;
+    if (defaultColor) {
+      return defaultColor;
     }
+    if (isDataLayer(layer)) {
+      const sortedAccessors: string[] = getSortedAccessors(
+        frame.datasourceLayers[layer.layerId] ?? layer.accessors,
+        layer
+      );
 
-    const dataLayer: XYDataLayerConfig = layer;
-    const sortedAccessors: string[] = getSortedAccessors(
-      frame.datasourceLayers[layer.layerId] ?? layer.accessors,
-      layer
-    );
+      const colorAssignments = getColorAssignments(
+        getDataLayers(state.layers),
+        { tables: frame.activeData },
+        formatFactory
+      );
+      const mappedAccessors = getAccessorColorConfig(
+        colorAssignments,
+        frame,
+        {
+          ...layer,
+          accessors: sortedAccessors.filter((sorted) => layer.accessors.includes(sorted)),
+        },
+        paletteService
+      );
 
-    const colorAssignments = getColorAssignments(
-      getDataLayers(state.layers),
-      { tables: frame.activeData },
-      formatFactory
-    );
-    const mappedAccessors = getAccessorColorConfig(
-      colorAssignments,
-      frame,
-      {
-        ...dataLayer,
-        accessors: sortedAccessors.filter((sorted) => dataLayer.accessors.includes(sorted)),
-      },
-      paletteService
-    );
+      return mappedAccessors.find((a) => a.columnId === accessor)?.color || null;
+    }
+  }, [
+    overwriteColor,
+    frame,
+    paletteService,
+    state.layers,
+    accessor,
+    formatFactory,
+    layer,
+    defaultColor,
+  ]);
 
-    return mappedAccessors.find((a) => a.columnId === accessor)?.color || null;
-  }, [overwriteColor, frame, paletteService, state.layers, accessor, formatFactory, layer]);
+  useEffect(() => {
+    setColor(currentColor);
+  }, [currentColor]);
 
   const [color, setColor] = useState(currentColor);
 
```

Comment:
No must have, but I think it's a little weird the default of 10% opacity is dropped in favor of 100% if a color is picked from the swatch. What do you think about keeping the previous alpha value? We can do this by passing in the `swatches` prop and setting it to `euiPaletteColorBlind()` with the opacity of the current color value.